library/common/http/header_utility.cc

```diff
@@ -65,6 +66,12 @@ envoy_headers toBridgeHeaders(const HeaderMap& header_map) {
 
         return HeaderMap::Iterate::Continue;
       });
+  if (!alpn.empty()) {
+    envoy_data key = Data::Utility::copyToBridgeData("x-envoy-upstream-alpn");
```

Comment:
Naïve question: is copying the String really needed?

Reply:
I think that @goaway had plans for perf improvements but AFIK this is currently the way it's done.

Reply:
Rather than augmenting the transformation call, I think it might be preferable to simply add the new header to Envoy's header map prior to transforming it to the bridge format.

Reply:
I suppose that's mostly a nit though - if you'd prefer to move forward, that's fine, too.

Reply:
yeah I prefer to move forward becasue as said if we add it to the header map we have to do a fully copy of the headers (the header map is const) before doing another full copy (to the shared memory struct)